test/e2e/cleanup/sweeper.go

```diff
@@ -80,89 +86,103 @@ func (c *Sweeper) Run(ctx context.Context, input SweeperInput) error {
-	}
-
-	if err := c.cleanupKeyPairs(ctx, filterInput); err != nil {
-		return fmt.Errorf("cleaning up key pairs: %w", err)
-	}
-
-	if err := c.cleanupSSMParameters(ctx, filterInput); err != nil {
-		return fmt.Errorf("cleaning up SSM parameters: %w", err)
-	}
-	if err := c.cleanupSSMManagedInstances(ctx, filterInput); err != nil {
-		return fmt.Errorf("cleaning up SSM managed instances: %w", err)
+	cleanups := []Cleanup{
+		{
+			Cleanup:        c.cleanupEC2Instances,
+			FailureMessage: "cleaning up EC2 instances",
+		},
+		{
+			Cleanup:        c.cleanupIAMInstanceProfiles,
+			FailureMessage: "cleaning up IAM instance profiles",
+		},
+		{
+			Cleanup:        c.cleanupCredentialStacks,
+			FailureMessage: "cleaning up credential stacks",
+		},
+		{
+			Cleanup:        c.cleanupEKSClusters,
+			FailureMessage: "cleaning up EKS clusters",
+		},
+		{
+			Cleanup:        c.emptyS3PodIdentityBuckets,
+			FailureMessage: "emptying S3 pod identity buckets",
+		},
+		{
+			Cleanup:        c.cleanupArchitectureStacks,
+			FailureMessage: "cleaning up architecture stacks",
+		},
+		{
+			Cleanup:        c.cleanupRolesAnywhereProfiles,
+			FailureMessage: "cleaning up Roles Anywhere profiles",
+		},
+		{
+			Cleanup:        c.cleanupRolesAnywhereTrustAnchors,
+			FailureMessage: "cleaning up Roles Anywhere trust anchors",
+		},
+		{
+			Cleanup:        c.cleanupIAMRoles,
+			FailureMessage: "cleaning up IAM roles",
+		},
+		{
+			Cleanup:        c.cleanupPeeringConnections,
+			FailureMessage: "cleaning up peering connections",
+		},
+		{
+			Cleanup:        c.cleanupInternetGateways,
+			FailureMessage: "cleaning up internet gateways",
+		},
+		{
+			Cleanup:        c.cleanupNetworkInterfaces,
+			FailureMessage: "cleaning up network interfaces",
+		},
+		{
+			Cleanup:        c.cleanupSubnets,
+			FailureMessage: "cleaning up subnets",
+		},
+		{
+			Cleanup:        c.cleanupRouteTables,
+			FailureMessage: "cleaning up route tables",
+		},
+		{
+			Cleanup:        c.cleanupSecurityGroups,
+			FailureMessage: "cleaning up security groups",
+		},
+		{
+			Cleanup:        c.cleanupVPCs,
+			FailureMessage: "cleaning up VPCs",
+		},
+		{
+			Cleanup:        c.cleanupS3PodIdentityBuckets,
+			FailureMessage: "cleaning up S3 pod identity buckets",
+		},
+		{
+			Cleanup:        c.cleanupKeyPairs,
+			FailureMessage: "cleaning up key pairs",
+		},
+		{
+			Cleanup:        c.cleanupSSMParameters,
+			FailureMessage: "cleaning up SSM parameters",
+		},
+		{
+			Cleanup:        c.cleanupSSMManagedInstances,
+			FailureMessage: "cleaning up SSM managed instances",
+		},
+		{
+			Cleanup:        c.cleanupSSMHybridActivations,
+			FailureMessage: "cleaning up SSM hybrid activations",
+		},
+	}
+
+	allErrors := []error{}
+	for _, cleanup := range cleanups {
+		if err := cleanup.Cleanup(ctx, filterInput); err != nil {
+			c.logger.Error(err, cleanup.FailureMessage)
+			allErrors = append(allErrors, fmt.Errorf("%s: %w", cleanup.FailureMessage, err))
```

Comment:
this will always fail in followings deletes when a dependent resource cant be deleted
are we gonna retry those deletes a lot of times or just 3?